### ISSUE TITLE
Script evaluation fails on ToString() on simple types

### ISSUE THREAD

**Issue by gunsh** [microsoft/ClearScript#173]
I have encountered a strange bug which relates to invoking ToString() on a NET object. This seems to work on everything except the simplest types (string, int, bool, double, char, etc). Please find the code that reproduces this in the following repo: https://github.com/gunsh/clearscript-to-string-bug

Here is the program output which shows success for DateTime, Uri, KeyValuePair<> and object as well as failure for string, int, double, bool and char.

I am guessing this is because simple types are converted to JavaScript internal types which makes them non-NET types but that makes simple tasks like "text".Substring(1) or 1234.ToString("N") impossible.

```
4/11/2020 2:03:18 PM
https://github.com/
[A, B]
System.Object
TypeError: context.Value.ToString is not a function Error testing String!!! [String]
TypeError: context.Value.ToString is not a function Error testing 0 [Int32]
TypeError: context.Value.ToString is not a function Error testing 3.14 [Double]
TypeError: context.Value.ToString is not a function Error testing True [Boolean]
TypeError: context.Value.ToString is not a function Error testing A [Char]
```

**Comment by ClearScriptLib** [comment]
Hello @gunsh,

>I am guessing this is because simple types are converted to JavaScript internal types which makes them non-NET types

That's correct. Without automatic conversion of those fundamental types, using ClearScript would be very painful.

>but that makes simple tasks like "text".Substring(1) or 1234.ToString("N") impossible.

There are many ways to provide access to such methods. One possibility is to expose a generic method that stores the value in a [_host variable_](https://microsoft.github.io/ClearScript/Reference/html/M_Microsoft_ClearScript_HostFunctions_newVar__1.htm):

``` C#
engine.Script.host = new HostFunctions();
engine.Execute(@"
    Object.prototype.toClr = function () {
        return host.newVar(this.valueOf());
    }
");
```

With this setup, you can make your test work as follows:

``` C#
Console.WriteLine(engine.Evaluate("context.Value.toClr().ToString()"));
```

You can also use expressions such as `"text".toClr().Substring(1)`. Unfortunately `1234.toClr().ToString("N")` is a syntax error in JavaScript.

Good luck!

**Comment by gunsh** [comment]
Thank you @ClearScriptLib. That was very helpful.

Fortunately (or not), I have rolled my own solution even before you managed to post yours. I have pushed an update to the sample repo to a ["fix" branch](https://github.com/gunsh/clearscript-to-string-bug/tree/fix).

I find this so useful perhaps it's worth including some solution for this problem into a future update of built-in `HostFunctions`?

What I did is essentially expose a `$` global function that wraps a NET object (especially simple types) into a wrapper that implements `DynamicObject`. This gives a super simple syntax that allows `Substring(1)` or `ToString("N")` or anything else to work.

`$(1000).ToString("N")` outputs `1,000.00`

`$("abcd").Substring(1)` outputs `bcd`
